### PR TITLE
New option: --outdir

### PR DESCRIPTION
**Pull request by vitasam**
New options have been added:
`--yuv        process RGB->YUV and save Y-plane (scaled to 0-255) to '.bin' file`
`----outdir   output directory for processed files; default = current dir`

### CODE REVIEW

File: bayer2ppm [hunk 1]
```diff
@@ -94,11 +97,49 @@ def bayer2ppm(config, infilespec):
...
+        t1Start = t1 = time.time()
+        print(f"Converting {infilename} to YUV...")
+        yuv_image = rgb_to_yuv(config, image.copy())
+        print(f"{_elapsed(t1)} - converting RGB to YUV")
+
+        # Get Y-plane and scale it to 0-255 range
+        y_plane = yuv_image[:,:,0]
+        y_plane *= (255.0 / y_plane.max())
+        y_plane_data = y_plane.astype(np.uint8)
+        np.clip(y_plane_data, 0, 255)
```
Comment (toaarnio) [inline bayer2ppm]: Clipping should be done before conversion to uint8. Doesn't make much sense to clip an 8-bit value to [0, 255]. :-)

Comment (vitasam) [reply bayer2ppm]: Oops. Lets assume it is a typo :)

File: bayer2ppm [hunk 2]
```diff
@@ -94,11 +97,49 @@ def bayer2ppm(config, infilespec):
...
+            print(f"{_elapsed(t1)} - scale Y-plane -> min: {y_plane_data.min()}, max: {y_plane_data.max()}")
+            plt.figure()
+            plt.title('Y-plane 8bits')
+            plt.imshow(y_plane_data, cmap = 'gray')
+            plt.colorbar()
+            print(f"{_elapsed(t1)} - plot Y-plane. Press 'q' to close the plot figure")
+            plt.show()
+
+        # Save Y-plane to a binary file
+        y_plane_filename = outfilename.rsplit('.', 1)[0] + '_y_plane.bin'
```
Comment (toaarnio) [inline bayer2ppm]: I'd suggest `os.path.splitext(outfilename)[0]` instead of `rsplit`.

File: bayer2ppm [hunk 3]
```diff
@@ -26,7 +26,7 @@ import imgio            # pip install imgio
 import rawpy            # pip install rawpy
 import rawpipe          # pip install rawpipe
 import argv             # local import: argv.py
-
+import matplotlib.pyplot as plt # pip install matplotlib
```
Comment (toaarnio) [inline bayer2ppm]: Perhaps use a separate tool for viewing the Y-data? I'd like to avoid having matplotlib as a dependency.

File: bayer2ppm [hunk 4]
```diff
@@ -234,9 +286,14 @@ def main():
     config.gammamode = argv.stringval("--gamma", default=None)
     config.verbose = not argv.exists("--quiet")
     config.debug = argv.exists("--debug")
+    config.process_yuv = argv.exists("--yuv")
+    config.outdir = argv.stringval("--outdir", default=None)
     show_help = argv.exists("--help")
     argv.exitIfAnyUnparsedOptions()
 
+    if config.verbose:
+        print('Bayer RAW convertor, ver %s\n' % config.version)
```
Comment (toaarnio) [inline bayer2ppm]: Suggest changing printf-style syntax to Python3 f-strings:
`print(f"Bayer RAW converter, version {config.version}")`

File: bayer2ppm (hunk 3 above)
Comment (vitasam) [reply bayer2ppm]: Yes, agree, let's remove this dependency. 

Meanwhile is `rawpy` really needed to be 0.14 or newer? My Windows 10's WSL Ubuntu seems does not have it out of the box: 

```
Collecting rawpy>=0.14.0 (from -r requirements.txt (line 5))
  Could not find a version that satisfies the requirement rawpy>=0.14.0 (from -r requirements.txt (line 5)) (from versions: 0.1.0, 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.3.4, 0.3.5, 0.4.0, 0.5.0, 0.5.1, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.9.0, 0.10.0a1, 0.10.0, 0.10.1, 0.11.0a1, 0.11.0, 0.12.0a1, 0.12.0, 0.13.0, 0.13.1)
No matching distribution found for rawpy>=0.14.0 (from -r requirements.txt (line 5))
```

**Review by toaarnio — CHANGES_REQUESTED** [review]
Thanks for the PR! I'd like a few things to be changed, though. Also a general question: Why not store the UV channels as well? Would that be adding too much complexity with all the variants like 4:0:0, 4:2:2, 4:2:0, interleaved vs. planar, etc?

**Comment by vitasam** [comment]
Maybe it is a good idea to store UV as well, at least to add such an option to select a format. The original idea of saving Y-plane only comes from the fact that the Y-plane is mostly used for CV needs.

**Comment by vitasam** [comment]
I've removed YUV part from this PR, `--outdir` parameter is the only change

File: bayer2ppm (hunk 3 above)
Comment (toaarnio) [reply bayer2ppm]: I think there was a bugfix in rawpy 0.14 that's required, can't recall what it was though. Version 0.15.0 is available via pip: https://pypi.org/project/rawpy/ and I have it on my WSL Ubuntu. Try `pip3 install --upgrade rawpy`.

**Pull request merged**